cmd/environment.go

```diff
@@ -142,19 +147,28 @@ func (env environment) LookupKMSConfig(config crypto.KMSConfig) (err error) {
 		}
 	}
 	if !config.Vault.IsEmpty() {
-		GlobalKMS, err = crypto.NewVault(config.Vault)
-		if err != nil {
-			return err
+		switch {
+		case config.Vault.Key.Name != "":
+			GlobalKMS, err = crypto.NewVaultKMS(config.Vault)
+			if err != nil {
+				return err
+			}
+			globalKMSKeyID = config.Vault.Key.Name
+		case config.Vault.KeyStorePath != "":
+			GlobalKeyStore, err = crypto.NewVaultKeyStore(config.Vault)
+			if err != nil {
+				return err
+			}
+		default:
 		}
-		globalKMSKeyID = config.Vault.Key.Name
 	}
 
 	autoEncryption, err := ParseBoolFlag(env.Get(EnvAutoEncryption, "off"))
 	if err != nil {
 		return err
 	}
 	globalAutoEncryption = bool(autoEncryption)
-	if globalAutoEncryption && GlobalKMS == nil { // auto-encryption enabled but no KMS
+	if globalAutoEncryption && (GlobalKMS == nil && GlobalKeyStore == nil) { // auto-encryption enabled but no KMS
 		return errors.New("Invalid KMS configuration: auto-encryption is enabled but no valid KMS configuration is present")
```

Comment:
should we have a generic error that includes the possibility of a missing key-value store configuration?